File: backend/src/main/java/com/joaomariajaneiro/neechathon/model/Product.java
package com.joaomariajaneiro.neechathon.model;

import javax.persistence.*;

@Entity
@Table(name = "PRODUCT")
public class Product {

    @Id
    @Column(name = "PRODUCT_ID")
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Column(name = "PRODUCT_NAME")
    private String name;

    @Column(name = "PRODUCT_PRICE")
    private Long price;

    @Column(name = "PRODUCT_DESCRIPTION")
    private String description;

    @Column(name = "PRODUCT_QUANTITY")
    private Integer quantity;

    @Column(name = "PRODUCT_IMAGE_PATH")
    private String image_path;

    public Product() {
    }

    public Product(String name, Long price, String description, Integer quantity, String image_path) {
        this.name = name;
        this.price = price;
        this.description = description;
        this.quantity = quantity;
        this.image_path = image_path;
    }

    public Long getId() {
        return id;
    }

    public String getDescription() {
        return description;
    }

    public Product setDescription(String description) {
        this.description = description;
        return this;
    }

    public Integer getQuantity() {
        return quantity;
    }

    public Product setQuantity(Integer quantity) {
        this.quantity = quantity;
        return this;
    }

    public String getImage_path() {
        return image_path;
    }

    public Product setImage_path(String image_path) {
        this.image_path = image_path;
        return this;
    }

    public Product setId(Long id) {
        this.id = id;
        return this;
    }

    public String getName() {
        return name;
    }

    public Product setName(String name) {
        this.name = name;
        return this;
    }

    public Long getPrice() {
        return price;
    }

    public Product setPrice(Long price) {
        this.price = price;
        return this;
    }
}

File: frontend/lib/src/backend_modulos/Classes/product.dart
class Product{
  int id;
  String name;
  int price;
  String description;
  int quantity;
  String image_path;
  Product(this.id,this.name, this.price, this.description, this.quantity,this.image_path);
  String toString() => '$id:  => $quantity  |  value = $price|  description = $description| name= $name|image= $image_path';
  Map toJson() => {'id': id, 'name':name, 'price': price, 'description': description, 'quantity': quantity, 'image_path':image_path};

  factory Product.fromJson(Map<String,dynamic> product){
    Product model_product= new Product(0, "", 0, "", 0, "");

    model_product.id=product["id"];
    model_product.name=product["name"];
    model_product.price=product["price"];
    model_product.description=product["description"];
    model_product.quantity=product["quantity"];
    model_product.image_path=product["image_path"];

    return model_product;
  }
}
